src/app/test_executive/gossip_consistency.ml

```diff
@@ -0,0 +1,87 @@
+open Core
+open Integration_test_lib
+open Currency
+open Signature_lib
+
+module Make (Inputs : Intf.Test.Inputs_intf) = struct
+  open Inputs
+  open Engine
+  open Dsl
+
+  (* TODO: find a way to avoid this type alias (first class module signatures restrictions make this tricky) *)
+  type network = Network.t
+
+  type node = Network.Node.t
+
+  type dsl = Dsl.t
+
+  let block_producer_balance = "1000" (* 1_000_000_000_000 *)
+
+  let config =
+    let n = 5 in
+    let open Test_config in
+    { default with
+      block_producers=
+        List.init n ~f:(fun _ ->
+            {Block_producer.balance= block_producer_balance; timing= Untimed}
+        ) }
+
+  let expected_error_event_reprs = []
+
+  let pk_of_keypair keypairs n =
+    Public_key.compress (List.nth_exn keypairs n).Keypair.public_key
+
+  let wait_for_all_to_initialize ~logger network t =
+    let open Malleable_error.Let_syntax in
+    let producers = Network.block_producers network in
+    let n = List.length producers in
+    List.mapi producers ~f:(fun i node ->
+        let%map () = wait_for t (Wait_condition.node_to_initialize node) in
+        [%log info] "Block producer %d (of %d) initialized" (i + 1) n ;
+        () )
+    |> Malleable_error.all_unit
+
+  let send_payments ~logger ~sender ~receiver ~node n =
+    let open Malleable_error.Let_syntax in
+    let amount = Currency.Amount.of_int 1 in
+    let fee = Currency.Fee.of_int 1 in
```

Comment:
is this fee sufficient? will it always be?

Reply:
turns out the answer is no lol